### PR TITLE
js: Update server, add test of updating a stream with sources

### PR DESCRIPTION
**Pull request by wallyqs**
Signed-off-by: Waldemar Quevedo <wally@synadia.com>

### CODE REVIEW

File: test/js_test.go [hunk 1]
```diff
@@ -2875,6 +2876,53 @@ func testJetStreamMirror_Source(t *testing.T, nodes ...*jsServer) {
...
+		}
+
+		got = len(updated.Config.Sources)
+		if got != expected {
+			t.Errorf("Expected %v, got: %v", expected, got)
+		}
+
+		got = len(updated.Sources)
+		if got != expected {
+			t.Errorf("Expected %v, got: %v", expected, got)
```
Comment (wallyqs) [inline test/js_test.go:2918]: On previous revision of the server I was getting 0 here, seems fixed after upgrading to latest in the go.mod:
```
js_test.go:2918: Expected 2, got: 0
```

File: test/js_test.go [hunk 2]
```diff
@@ -2845,20 +2845,21 @@ func testJetStreamMirror_Source(t *testing.T, nodes ...*jsServer) {
...
 					time.Sleep(100 * time.Millisecond)
-					continue
+					continue Retry
 				}
 				msgs = append(msgs, msg)
 				continue GetNextMsg
 			}
 			if err != nil {
-				t.Fatalf("Unexpected error: %v", err)
+				t.Fatalf("Unexpected error fetching seq=%v: %v", seq, err)
```
Comment (wallyqs) [inline test/js_test.go:2862]: This is starting to flap a bit, waits 5 sec for a sequence to be present:
```
    js_test.go:2862: Unexpected error fetching seq=12: stream store EOF
```

Comment (derekcollison) [reply test/js_test.go:2862]: Next large PR should fix a bunch of those.

**Review by derekcollison — APPROVED** [review]
LGTM

**Pull request merged**